Add a 0.2 release blog post (#679)
unicode-org/icu4x: docs/posts/20210427_ICU4X_02_Release.md

# ICU4X 0.2 Released

> Unicode® ICU4X 0.2 has just been released.

Reviewer: Suggestion: avoid use of passive voice here.
```suggestion
We are excited to announce the release of Unicode® ICU4X 0.2.
```

> The revision improves completness of the components introduced in ICU4X 0.1 and introduces a number of lower level utilities.

Reviewer: ```suggestion
The revision improves completeness of the components introduced in ICU4X 0.1 and introduces a number of lower level utilities.
```

## Components

> In 0.2, ICU4X improves `DateTimeFormat` adding initial support for time zones, component bags and day periods. A new `LocaleCanonicalizer` component has been added to support `Locale` data driven operations. Third improved component is `PluralRules` which gained support for the `E` operand.

Reviewer 2: And FixedDecimalFormat? :)

> In 0.2, ICU4X improves `DateTimeFormat` adding initial support for time zones, component bags and day periods. A new `LocaleCanonicalizer` component has been added to support `Locale` data driven operations. Third improved component is `PluralRules` which gained support for the `E` operand.

Reviewer: ```suggestion
In 0.2, ICU4X improves `DateTimeFormat` adding initial support for time zones, component bags and day periods. A new `LocaleCanonicalizer` component has been added to support `Locale` data driven operations. A third improved component is `PluralRules` which has gained support for the `E` operand.
```

> 

Reviewer 2: Also, please give a shout out to the C FFI.

## Utilities

> With a small set of core components maturing, the project starts to focus more on the features we want to develop for all components such as strong FFI support and flexible data management.

Reviewer: ```suggestion
With a small set of core components maturing, the project is starting to focus more on the features we want to develop for all components such as strong FFI support and flexible data management.
```

> For that purpose, we started building a number of lower level utilities focused on bringing high performance data structures and algorithms.

Reviewer: ```suggestion
For that purpose, we have started building a number of lower level utilities focused on bringing high performance data structures and algorithms.
```

> To minimize the size of our WASM binaries we added `Writeable`, which introduces a lightweight alternative to `Display` and `LiteMap`, which adds a `HashMap` replacement that uses `binary_search` instead of carrying with it a hashing logic.

Reviewer: ```suggestion
To minimize the size of our WASM binaries we added `Writeable`, which introduces a lightweight alternative to `Display` and `LiteMap`, which adds a `HashMap` replacement that uses `binary_search` instead of a hash function.
```

## FFI

ICU4X FFI bindings include a new `C` API prototype. While not part of the release itself, it serves as a validation of the core API model and the team expects it to be published alongside ICU4X in the next release.

## Utilities

With a small set of core components maturing, the project is starting to focus more on the features we want to develop for all components such as strong FFI support and flexible data management.

For that purpose, we have started building a number of lower level utilities focused on bringing high performance data structures and algorithms.

To minimize the size of our WASM binaries we added `Writeable`, which introduces a lightweight alternative to `Display` and `LiteMap`, which adds a `HashMap` replacement that uses `binary_search` instead of a hash function.
`ZeroVec` is a utility similar to `Cow` which allows us to bring zero-copy deserialization of lists.
`FixedDecimal` and `Pattern` are two utilities that introduce lower level simple operations on numbers and ICU patterns.

### Summary

> ICU4X 0.2 comes with more feature-complete components, new utilities and CLDR 38.1 support.

Reviewer 2: We're on 39 now.
